tracer/src/Datadog.Trace/OTelMetrics/MetricReaderHandler.cs

```diff
@@ -0,0 +1,340 @@
+        {
+            if (state is MetricState metricState)
+            {
+                metricState.RecordMeasurementDouble(value, tags);
+            }
+            else
+            {
+                // Log error - this should not happen if we follow OTel SDK pattern
+                Log.Warning("Measurement recorded for instrument '{InstrumentName}' but no MetricState provided. Measurement will be dropped.", instrument.Name);
+            }
+        }
+
+        // Overloads for different numeric types
+        public static void OnMeasurementRecordedByte(Instrument instrument, byte value, ReadOnlySpan<KeyValuePair<string, object?>> tags, object? state)
+        {
+            OnMeasurementRecordedLong(instrument, value, tags, state);
+        }
+
+        public static void OnMeasurementRecordedShort(Instrument instrument, short value, ReadOnlySpan<KeyValuePair<string, object?>> tags, object? state)
+        {
+            OnMeasurementRecordedLong(instrument, value, tags, state);
+        }
+
+        public static void OnMeasurementRecordedInt(Instrument instrument, int value, ReadOnlySpan<KeyValuePair<string, object?>> tags, object? state)
+        {
+            OnMeasurementRecordedLong(instrument, value, tags, state);
+        }
+
+        public static void OnMeasurementRecordedFloat(Instrument instrument, float value, ReadOnlySpan<KeyValuePair<string, object?>> tags, object? state)
+        {
+            OnMeasurementRecordedDouble(instrument, value, tags, state);
+        }
+
+        internal static IReadOnlyList<MetricPoint> GetMetricsForExport(bool clearAfterGet = false)
+        {
+            var allMetricPoints = new List<MetricPoint>();
+
+            foreach (var kvp in CapturedMetrics)
+            {
+                var metricState = kvp.Value;
+                var metricPoints = metricState.GetMetricPoints();
+                allMetricPoints.AddRange(metricPoints);
+            }
+
+            if (clearAfterGet)
+            {
+                CapturedMetrics.Clear();
+            }
+
+            return allMetricPoints;
+        }
+
+        // For testing only
+        internal static IReadOnlyList<MetricPoint> GetCapturedMetricsForTesting()
+        {
+            var allMetricPoints = new List<MetricPoint>();
+
+            foreach (var kvp in CapturedMetrics)
+            {
+                var metricState = kvp.Value;
+                var metricPoints = metricState.GetMetricPoints();
+                allMetricPoints.AddRange(metricPoints);
+            }
+
+            return allMetricPoints;
+        }
+
+        // For testing only - reset all captured metrics
+        internal static void ResetForTesting()
+        {
+            CapturedMetrics.Clear();
+            MetricStreamNames.Clear();
+            Interlocked.Exchange(ref _metricCount, 0);
+        }
```

Comment:
nit: this is another sign for me that this whole type shouldn't be static. This is the sort of thing that's super easy to test and reason about if you can just do `new MetricReaderHandler()`. It also can make testing faster, as you can run all those tests in parallel if desired.

Reply:
Will remove static methods in next PR: https://github.com/DataDog/dd-trace-dotnet/pull/7514